examples/hosts/app-integration/external-controller/src/kvpair-dataobject/containerCode.ts

```diff
@@ -3,22 +3,31 @@
  * Licensed under the MIT License.
  */
 
-import { ContainerRuntimeFactoryWithDefaultDataStore } from "@fluidframework/aqueduct";
+import { BaseContainerRuntimeFactory } from "@fluidframework/aqueduct";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { innerRequestHandler, RuntimeRequestHandler } from "@fluidframework/request-handler";
+import { RequestParser } from "@fluidframework/runtime-utils";
 import { KeyValueInstantiationFactory } from "./DataObject";
 
 /**
- * The KeyValueContainerRuntimeFactory is the container code for our scenario.
- *
- * Since we only need to instantiate and retrieve a single dice roller for our scenario, we can use a
- * ContainerRuntimeFactoryWithDefaultDataStore. We provide it with the type of the data object we want to create
- * and retrieve by default, and the registry entry mapping the type to the factory.
+ * We'll allow root data stores to be created by requesting a url like /create/dropletType/dataStoreId
+ */
+const createRequestHandler: RuntimeRequestHandler =
+    async (request: RequestParser, runtime: IContainerRuntime) => {
+        if (request.pathParts[0] === "create" && request.pathParts.length === 3) {
+            await runtime.createRootDataStore(request.pathParts[1], request.pathParts[2]);
+            return runtime.request(request.createSubRequest(2));
+        }
+    };
+
```

Comment:
Should this request be in aqueduct on the dataobject?

Also, could we have thepath be the same for both requests, but have create be a header param?

That way get & create could follow the same path.

Reply:
As far as I know the request must be handled by the container (unless we did an anchor-style root data object or similar, which we could consider).

Header could be fine too -- this is somewhat of an implementation detail since the goal here is to hide the request semantic from the consumer, let me know if you feel strongly.